pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap

```diff
@@ -3,999 +3,999 @@ source: pdg/src/main.rs
-	n[6]:  field.0     n[1] => _12  @ bb2[25]: fn invalid;
-	n[7]:  addr.load   n[6] => _    @ bb2[25]: fn invalid;
-	n[8]:  copy        n[1] => _23  @ bb4[13]: fn invalid;
-	n[9]:  copy        n[8] => _22  @ bb4[14]: fn invalid;
-	n[10]: free        n[9] => _21  @ bb4[15]: fn invalid;
+	n[0]:  alloc       _    => _2   @ bb1[2]:  fn invalid; _2 = pointers::malloc(move _3)            
+	n[1]:  copy        n[0] => _1   @ bb2[2]:  fn invalid; _1 = move _2 as *mut pointers::S (Misc)   
+	n[2]:  field.0     n[1] => _    @ bb2[5]:  fn invalid; ((*_1).0: i32) = const 10_i32             
+	n[3]:  addr.store  n[2] => _    @ bb2[5]:  fn invalid; ((*_1).0: i32) = const 10_i32             
+	n[4]:  copy        n[1] => _5   @ bb2[8]:  fn invalid; _5 = _1                                   
+	n[5]:  value.store n[4] => _6.* @ bb2[11]: fn invalid; (*_6) = move _5                           
+	n[6]:  field.0     n[1] => _12  @ bb2[25]: fn invalid; _12 = ((*_1).0: i32)                      
+	n[7]:  addr.load   n[6] => _    @ bb2[25]: fn invalid; _12 = ((*_1).0: i32)                      
+	n[8]:  copy        n[1] => _23  @ bb4[13]: fn invalid; _23 = _1                                  
+	n[9]:  copy        n[8] => _22  @ bb4[14]: fn invalid; _22 = move _23 as *mut libc::c_void (Misc)
+	n[10]: free        n[9] => _21  @ bb4[15]: fn invalid; _21 = pointers::free(move _22)            
 }
 nodes_that_need_write = [3, 2, 1, 0]
 
 g {
-	n[0]: addr.store _ => _ @ bb2[10]: fn invalid;
+	n[0]: addr.store _ => _ @ bb2[10]: fn invalid; (*_6) = move _5
 }
 nodes_that_need_write = [0]
 
 g {
-	n[0]: &_11 _    => _10 @ bb2[20]: fn invalid;
-	n[1]: copy n[0] => _9  @ bb2[21]: fn invalid;
-	n[2]: copy n[1] => _8  @ bb2[23]: fn invalid;
-	n[3]: copy n[2] => _1  @ bb0[0]:  fn printf; 
+	n[0]: &_11 _    => _10 @ bb2[20]: fn invalid; _10 = &raw const (*_11)                             
+	n[1]: copy n[0] => _9  @ bb2[21]: fn invalid; _9 = move _10 as *const u8 (Pointer(ArrayToPointer))
+	n[2]: copy n[1] => _8  @ bb2[23]: fn invalid; _8 = move _9 as *const i8 (Misc)                    
+	n[3]: copy n[2] => _1  @ bb0[0]:  fn printf; _7 = pointers::printf(move _8, move _12)             
 }
 nodes_that_need_write = []
 
 g {
-	n[0]: &_17 _    => _16 @ bb3[11]: fn invalid;
-	n[1]: copy n[0] => _15 @ bb3[12]: fn invalid;
-	n[2]: copy n[1] => _14 @ bb3[14]: fn invalid;
-	n[3]: copy n[2] => _1  @ bb0[0]:  fn printf; 
+	n[0]: &_17 _    => _16 @ bb3[11]: fn invalid; _16 = &raw const (*_17)                              
+	n[1]: copy n[0] => _15 @ bb3[12]: fn invalid; _15 = move _16 as *const u8 (Pointer(ArrayToPointer))
+	n[2]: copy n[1] => _14 @ bb3[14]: fn invalid; _14 = move _15 as *const i8 (Misc)                   
+	n[3]: copy n[2] => _1  @ bb0[0]:  fn printf; _13 = pointers::printf(move _14, move _18)            
 }
 nodes_that_need_write = []
 
 g {
-	n[0]: field.0 _ => _18 @ bb3[18]: fn invalid;
+	n[0]: field.0 _ => _18 @ bb3[18]: fn invalid; _18 = ((*(*_19)).0: i32)
```

Comment:
I think the input to this node should be a `LoadValue`.  This statement loads a pointer value `p` from `*_19`, then computes `q = &(*p).0`, and finally loads from `*q`.  So there should be two chains, `_19 -> LoadAddr` and `LoadValue -> Field -> LoadAddr`.

Also, the node having destination `_18` is incorrect.  Given the RHS, the type of `_18` must be `i32` (i.e. not a pointer type), so the PDG should not track anything about it.